### PR TITLE
[Personal Cabinet] - Admin - Display parent/child list

### PR DESCRIPTION
**Pull request by perepichai**
Task https://github.com/ita-social-projects/OoS-Frontend/issues/1105

### CODE REVIEW

File: src/app/shared/enum/tech-admin/tabs/users.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum userRole {
+  all = 'all',
+  parent = 'parent',
+  child = 'child'
+}
```
Comment (litvinets) [inline src/app/shared/enum/tech-admin/tabs/users.ts:5]: maybe you can move it to role.ts and add it to enum Role?

File: src/app/shared/utils/utils.ts [hunk 2]
```diff
@@ -105,4 +107,26 @@ export class Util {
...
+   */
+  public static updateStructureForTheTable(users): UsersTable[] {
+  const constants: typeof Constants = Constants;
+  let updatedUsers = [];
+  users.forEach((user) => {
+    updatedUsers.push({
+      id: user.id,
+      pib: `${user.lastName} ${user.firstName} ${user.middleName}` || `не вказано`,
+      email: user.email || 'не вказано',
+      place: user.place || 'не вказано',
```
Comment (litvinets) [inline src/app/shared/utils/utils.ts:124]: let's have constant noInformation='не вказано'

File: src/app/shell/admin-tools/admin-tools-routing.module.ts [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
   {
     path: 'platform/:index', component: PlatformComponent,
     loadChildren: () => import('./platform/platform.module').then(m => m.PlatformModule),
+  },
+  {
+    path: 'users', component: UsersComponent,
+  },
+  {
+    path: 'users/:index', component: UsersComponent,
   }
```
Comment (litvinets) [inline src/app/shell/admin-tools/admin-tools-routing.module.ts:20]: use one path users/:index
and pass empty string for example if there is no param

File: src/app/shell/admin-tools/users/users.component.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+})
+export class UsersComponent implements OnInit {
+
+  @Select(UserState.isLoading)
+  isLoadingCabinet$: Observable<boolean>;
+  @Select(AdminState.children)
+  children$: Observable<ChildCards>;
+
+  readonly userRoleUkr = UserTabsUkr;
+  readonly noUsers = NoResultsTitle.noUsers;
```
Comment (litvinets) [inline src/app/shell/admin-tools/users/users.component.ts:30]: move readonly to the very begin as in the rest of the files:)

File: src/app/shell/admin-tools/users/users.component.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+  filter = new FormControl('');
+  filterValue: string;
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  tabIndex: number;
+  allUsers = [];
+  parents = [];
```
Comment (litvinets) [inline src/app/shell/admin-tools/users/users.component.ts:37]: add Typo

File: src/app/shell/admin-tools/users/users.component.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+  constructor(
+    public store: Store,
+    private router: Router,
+    private route: ActivatedRoute,
+  ) { }
+
+  ngOnInit(): void {
+    this.filter.valueChanges
+    .pipe(takeUntil(this.destroy$), debounceTime(200), distinctUntilChanged())
```
Comment (litvinets) [inline src/app/shell/admin-tools/users/users.component.ts:48]: I think it's not possible to have !val not an empty string (I mean is it possible to receive here null/undefined/0/NaN ??? ), right? So you can simplify it to
this.filterValue = val;

**Review by dmitryy90 — APPROVED** [review]

**Review by litvinets — APPROVED** [review]
Fix please comments and DO NOT merge for now. let's wait until this PR will be merged  https://github.com/ita-social-projects/OoS-Frontend/pull/1181

**Pull request merged**